### PR TITLE
batch login node supports multiple batch jobs

### PR DESCRIPTION
**Pull request by heyealex**
## Description
Update the variables to support passing in a list of batch jobs, or using a list of batch jobs modules, with the Batch login node module. All job templates will be placed in the same directory and instructions for executing the job templates will be provided for all jobs. The first provided job template will define the shape of the login node and it's environment.

An example of how 2 jobs can be provided to the same login node can be found in the new test_config named `multiple-batch-jobs.yaml`.

This PR won't be submitted until after #642 and conflicts have been resolved.

### Deprecations
This PR merges 3 variables previously supplied separates to simplify validation and implementation. They are still used if provided, but the output of `batch-job-template` doesn't provide them anymore and they have been labeled as deprecated.

* `job_filename`
* `job_id`
* `job_template_contents`

These are replaced with:
```hcl
job_data = object({
    template_contents = string,
    filename          = string,
    id                = string
})
```

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

**Review by tpdownes — APPROVED** [review]
The choice to prepend the deprecated variables seems like a reasonable one. LGTM.

**Comment by heyealex** [comment]
I had a couple formatting fixes come up after the integration test kicked off. Its functionally still the same, so that test should still be valid.

File: community/modules/scheduler/cloud-batch-login-node/README.md [hunk 1]
```diff
@@ -6,10 +6,11 @@ module.
 
 This login node:
 
-- Uses the same VM settings as the `cloud-batch-job`, such as image, machine type, etc...
-- Has the same mounted file systems as the `cloud-batch-job`.
-- Runs the same `startup-script` as the `cloud-batch-job`.
-- Contains a folder with the job template generated by the `cloud-batch-job`.
+- Uses the same VM settings as the first provided `cloud-batch-job`, such as
+  image, machine type, etc...
+- Has the same mounted file systems as the first provided `cloud-batch-job`.
```
Comment (nick-stroud) [inline community/modules/scheduler/cloud-batch-login-node/README.md:11]: Does it really only mount file systems from the first?

File: community/modules/scheduler/cloud-batch-login-node/main.tf [hunk 2]
```diff
@@ -32,19 +46,20 @@ locals {
...
   Delete job:
-    gcloud ${var.gcloud_version} batch jobs delete ${var.job_id} --location=${var.region} --project=${var.project_id}
+    gcloud ${var.gcloud_version} batch jobs delete ${job.id} --location=${var.region} --project=${var.project_id}
 
-  List all jobs:
-    gcloud ${var.gcloud_version} batch jobs list --project=${var.project_id}
   EOT
+  ])
 
   readme_contents = <<-EOT
```
Comment (nick-stroud) [inline community/modules/scheduler/cloud-batch-login-node/main.tf:64]: nit: now readme contents will not contain the List all jobs command. Probably worth including but not critical.

File: community/modules/scheduler/cloud-batch-login-node/variables.tf [hunk 3]
```diff
@@ -59,23 +63,48 @@ variable "startup_script" {
...
+  }
+  validation {
+    condition     = length(distinct([for job in var.job_data : job.id])) == length(var.job_data)
+    error_message = "All job IDs in var.job_data must be unique."
+  }
+}
+
 variable "job_template_contents" {
-  description = "The contents of the Google Cloud Batch job template. Typically supplied by a cloud-batch-job module."
+  description = "Deprecated: The contents of the Google Cloud Batch job template. Typically supplied by a cloud-batch-job module."
```
Comment (nick-stroud) [inline community/modules/scheduler/cloud-batch-login-node/variables.tf:84]: nit: On the deprecated variables, I would say _"Deprecated (use `job_data`):"_.

File: community/modules/scheduler/cloud-batch-login-node/main.tf (hunk 2 above)
Comment (heyealex) [reply community/modules/scheduler/cloud-batch-login-node/main.tf:64]: It still does include the list all jobs command - it just moved to the top. This is the output from the test_config:

```
Batch job template files will be placed on the Batch login node in the following directory:
          /home/batch-jobs
        
        Use the following commands to:
        SSH into the login node:
          gcloud compute ssh --zone us-central1-c hello-workload-batch-login  --project gcp-hpc-demo-0
        
        List all jobs:
          gcloud alpha batch jobs list --project=gcp-hpc-demo-0
        
        ## For job: hello-workload ##
        
        Submit your job from login node:
          gcloud alpha batch jobs submit hello-workload --config=/home/batch-jobs/cloud-batch-hello-workload.json --location=us-central1 --project=gcp-hpc-demo-0
          
        Check status:
          gcloud alpha batch jobs describe hello-workload --location=us-central1 --project=gcp-hpc-demo-0 | grep state:
          
        Delete job:
          gcloud alpha batch jobs delete hello-workload --location=us-central1 --project=gcp-hpc-demo-0
        
        
        ## For job: job-2 ##
        
        Submit your job from login node:
          gcloud alpha batch jobs submit job-2 --config=/home/batch-jobs/batch-job-2.json --location=us-central1 --project=gcp-hpc-demo-0
          
        Check status:
          gcloud alpha batch jobs describe job-2 --location=us-central1 --project=gcp-hpc-demo-0 | grep state:
          
        Delete job:
          gcloud alpha batch jobs delete job-2 --location=us-central1 --project=gcp-hpc-demo-0
```

File: community/modules/scheduler/cloud-batch-login-node/README.md (hunk 1 above)
Comment (heyealex) [reply community/modules/scheduler/cloud-batch-login-node/README.md]: Not any more, updated.

File: community/modules/scheduler/cloud-batch-login-node/variables.tf (hunk 3 above)
Comment (heyealex) [reply community/modules/scheduler/cloud-batch-login-node/variables.tf]: Good point, updated.

File: community/modules/scheduler/cloud-batch-login-node/main.tf (hunk 2 above)
Comment (nick-stroud) [reply community/modules/scheduler/cloud-batch-login-node/main.tf:64]: I meant the readme that is placed on the login node by the data runner ([ref](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/372fe04b7d7a37dba5d5027d11b4d5c6680f1bc4/community/modules/scheduler/cloud-batch-login-node/main.tf#L107)). I believe the text you posted above is the instructions output.

**Review by nick-stroud — APPROVED** [review]

File: community/modules/scheduler/cloud-batch-login-node/main.tf (hunk 2 above)
Comment (heyealex) [reply community/modules/scheduler/cloud-batch-login-node/main.tf]: Understood, I got it confused with the instructions. I converted the line to a local var and used in both.

**Review by nick-stroud — APPROVED** [review]

**Pull request merged**